core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackService.java

```diff
@@ -493,29 +493,42 @@ private void stop(Stack stack, Cluster cluster, boolean updateCluster) {
     }
 
     private void triggerStackStopIfNeeded(Stack stack, Cluster cluster, boolean updateCluster) {
+        if (!isStopNeeded(stack)) {
+            return;
+        }
+        if ((cluster != null && !cluster.isStopped()) && !stack.isStopFailed()) {
+            if (!updateCluster) {
```

Comment:
`updateCluster` must be true, so would be nice to check this one on the caller side

Reply:
We used to support events on the stack without cluster modifications, so it validates when the stack has a cluster and it is not in stopped state then it could not be stopped, doesn't it?
I think we could keep this validation as it is and covering/prune this validation in a separate ticket.